extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -382,17 +382,21 @@ public void configurationDescriptorBuilding(
         // First produce the PUs having a persistence.xml: these are not reactive, as we don't allow using a persistence.xml for them.
         for (PersistenceXmlDescriptorBuildItem persistenceXmlDescriptorBuildItem : persistenceXmlDescriptors) {
             ParsedPersistenceXmlDescriptor xmlDescriptor = persistenceXmlDescriptorBuildItem.getDescriptor();
+            Optional<JdbcDataSourceBuildItem> jdbcDataSource = jdbcDataSources.stream()
+                    .filter(i -> i.isDefault())
+                    .findFirst();
```

Comment:
This confuses me a bit - why the first one?

Reply:
No particular reason, `findAny` (if that exists) would work. I copy pasted that code from the method that handles non-persistence.xml config.

Reply:
But don't we attempt to match the right datasource configuration for this PU ?

Reply:
The code for `persistence.xml` handling has always assumed we were using the default datasource for these PUs:

https://github.com/yrodiere/quarkus/blob/8f171cdbcfa4c810f5ec10377240a5f3278607c0/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java#L396-L399

This new code is in line with that assumption.

If you think it's wrong and we should extract the datasource name from the `persistence.xml` (I guess it can be configured there?), I can open another ticket?

Reply:
ok, existing code - no rush then. But yes it surprised me a bit - I guess I forgot we ignore the datasource section from a persistence.xml.
I guess ignoring it is fine, but to make things better we could try to spot if the user defined a datasource section and warn about it being ignored / or even fail.